src/pycram/language.py

```diff
@@ -260,34 +261,49 @@ def __init__(self, condition: Union[Callable, Fluent] = None):
         """
         super().__init__(None, None)
         self.kill_event = threading.Event()
+        self.exception_queue = queue.Queue()
```

Comment:
Are you sure this shouldbe a queue? Python Queue does not mean queue, but synchronized queue. queue.dequeue is the asynchronous way i think.

Reply:
the children are running in different threads, and as i understand queue.Queue is threadsafe, and synchronized queue makes sense here? I could be wrong though, input appreciated.
Also i looked up queue.dequeue, but only found queue.deque, but that does not seem to be the asynchronous way.

Reply:
Cool, then its correct.